### DISCUSSION TITLE
ec.rotate_elements inverts geometry when rotation axis vector has zero-length

### DISCUSSION THREAD

**Discussion by roehligas** [cwapi3d/cwapi3dpython#161] [General] [answered]
Hi,

When elements are rotated around a vector with 0 magnitude (for example [0,0,0] or in my case [0,-0,0]), the geometry gets inverted/destroyed.

Are there use cases where this behavior is desired? If not I suggest throwing an Error when the rotation axis has zero-length.
Took me a very long time to figure out where the problem actually was coming from because it randomly fell together with the desired rotation direction for my use case, so the elements got turned but were broken after...

```

import element_controller as ec
import cadwork as cw
import math

active_elements= ec.get_active_identifiable_element_ids()
ec.rotate_elements(active_elements, cw.point_3d(0,0,0), cw.point_3d(0,0,0), math.pi)

```

![image](https://github.com/cwapi3d/cwapi3dpython/assets/66385500/84c91aca-2805-475c-b383-38f19aa584a3)
![image](https://github.com/cwapi3d/cwapi3dpython/assets/66385500/341a5601-3c66-4eae-8e90-df0853b58663)
![image](https://github.com/cwapi3d/cwapi3dpython/assets/66385500/c9439dfa-f802-4ec5-b164-90fac7b9be10)

**Comment by Brunner246** [comment, marked as answer]
Thank you for reporting the problem. We will enforce an invariant that the rotation axis vector should never have a zero magnitude. This will prevent such issues.
